lib/membrane_dashboard_web/live/dashboard_live.ex

```diff
@@ -84,74 +84,97 @@ defmodule Membrane.DashboardWeb.DashboardLive do
-          time_from: from,
-          time_to: to,
           accuracy: accuracy,
           update: update,
           update_range: update_range
         )
 
       {:noreply, socket}
     else
-      {:error, reason} ->
-        Logger.error(reason)
-        {:noreply, plan_update(socket)}
-
-      _ ->
-        {:noreply, plan_update(socket)}
+      something ->
+        Logger.error("Encountered invalid arguments when handling params: #{inspect(something)}")
+        {:noreply, socket}
     end
   end
 
   # inits, realoads or updates charts and reloads dagre
   @impl true
-  def handle_info({:charts_init, data}, socket),
-    do: {:noreply, push_event(socket, "charts_init", %{data: data})}
+  def handle_info({:charts_init, data}, socket) do
+    {:noreply, push_event(socket, "charts_init", %{data: data})}
+  end
+
+  def handle_info({event, charts, data, paths}, socket)
+      when event in [:charts_data, :charts_update] do
+    socket =
+      socket
+      |> push_event(Atom.to_string(event), %{data: charts})
+      |> assign(data: data, paths: paths)
+
+    {:noreply, socket}
+  end
+
+  def handle_info({:alive_pipelines, alive_pipelines}, socket) do
+    {:noreply, assign(socket, alive_pipelines: alive_pipelines)}
+  end
+
+  # this message is the last message sent from query task
+  def handle_info({:update_query_time, time_from, time_to}, socket) do
+    socket =
+      socket
+      |> assign(time_from: time_from, time_to: time_to)
+      |> case do
+        # if the timer had not been set then do nothing
+        %{assigns: %{update_ref: nil}} = socket ->
+          socket
 
-  def handle_info({:charts_data, data}, socket),
-    do: {:noreply, push_event(socket, "charts_data", %{data: data})}
+        socket ->
+          socket |> plan_update()
+      end
 
-  def handle_info({:charts_update, data}, socket),
-    do: {:noreply, push_event(socket, "charts_update", %{data: data})}
+    {:noreply, socket}
+  end
 
-  def handle_info({:dagre_data, data}, socket),
-    do: {:noreply, push_event(socket, "dagre_data", %{data: data})}
+  def handle_info({:dagre_data, data}, socket) do
+    {:noreply, push_event(socket, "dagre_data", %{data: data})}
+  end
+
+  def handle_info({:set_data_loading, loading}, socket) do
+    {:noreply, assign(socket, data_loading: loading)}
+  end
 
   def handle_info(:update, socket) do
-    if socket.assigns.update and socket.assigns.next_update_time <= now() do
-      socket = assign(socket, next_update_time: now(@min_between_updates_interval))
+    {:noreply,
+     push_patch_with_params(socket, %{
+       mode: :update,
+       from: now(-socket.assigns.update_range),
+       to: now()
+     })}
+  end
 
-      {:noreply,
-       push_patch_with_params(socket, %{
-         mode: :update,
-         from: now(-socket.assigns.update_range),
-         to: now()
-       })}
-    else
-      {:noreply, socket}
-    end
+  def handle_info(
+        {:DOWN, query_task_ref, :process, _pid, reason},
```

Comment:
why atom `:DOWN` has big letters?

Reply:
It is erlang's thing. In elixir/erlang we have a process monitoring capabilities and we can monitor when process goes down by using [Process.monitor/1](https://hexdocs.pm/elixir/1.12/Process.html#monitor/1). So basically you can detect when process created via `spawn` went down and you will receive a `:DOWN` message.

Reply:
okay, thanks :D